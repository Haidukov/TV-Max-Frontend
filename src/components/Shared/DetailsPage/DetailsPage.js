/* eslint-disable react/no-danger */
import React from 'react';
import PropTypes from 'prop-types';

import s from './DetailsPage.scss';
import defaultImageSrc from '../../../assets/images/no-img-landscape-text.png';

const DetailsPage = ({ name, summary, imageUrl }) => (
  <main data-testid='details'>
    <h1>{name}</h1>
    <div className={s.details}>
      <img src={imageUrl} className={s.logo} alt={name} />
      <div dangerouslySetInnerHTML={{ __html: summary }} />
    </div>
  </main>
);

DetailsPage.propTypes = {
  name: PropTypes.string,
  summary: PropTypes.string,
  imageUrl: PropTypes.string
};

DetailsPage.defaultProps = {
  name: '',
  summary: '',
  imageUrl: defaultImageSrc
};

export default DetailsPage;
