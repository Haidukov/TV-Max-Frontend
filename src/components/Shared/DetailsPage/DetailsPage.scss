@import '../../../assets/styles/variables.scss';

.details {
    display: flex;
    @media screen and (max-width: 600px) {
        flex-direction: column;
    }
}

.logo {
    margin-right: 15px;
}