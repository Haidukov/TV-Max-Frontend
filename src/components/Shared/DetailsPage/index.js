export { default } from './DetailsPage.scss';
