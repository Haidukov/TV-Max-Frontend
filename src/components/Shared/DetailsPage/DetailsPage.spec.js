import React from 'react';
import { render } from '@testing-library/react';
import DetailsPage from './DetailsPage';

describe('<DetailsPage />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <DetailsPage
        summary='<div>summary</div>'
        name='test name'
        imageUrl='http://test.com/image.jpg'
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
