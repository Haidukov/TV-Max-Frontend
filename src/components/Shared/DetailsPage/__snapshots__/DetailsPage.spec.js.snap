// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsPage /> should match snapshot 1`] = `
<DocumentFragment>
  <main
    data-testid="details"
  >
    <h1>
      test name
    </h1>
    <div
      class="details"
    >
      <img
        alt="test name"
        class="logo"
        src="http://test.com/image.jpg"
      />
      <div>
        <div>
          summary
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
